static/app/views/settings/dynamicSampling/utils/rebalancing.tsx

```diff
@@ -0,0 +1,80 @@
+interface BalancingItem {
+  count: number;
+  id: string;
+  sampleRate: number;
+}
+
+interface Params<T extends BalancingItem> {
+  items: T[];
+  targetSampleRate: number;
+  intensity?: number;
+  minBudget?: number;
+}
+
+/**
+ * Balances the sample rate of items to match the target sample rate.
+ * Mirrors the behavior of the dynamic sampling backend.
+ *
+ * See `src/sentry/dynamic_sampling/models/projects_rebalancing.py`
+ * and `src/sentry/dynamic_sampling/models/full_rebalancing.py`
+ *
+ * @param targetSampleRate The target sample rate to balance the items to.
+ * @param items The items to balance.
+ * @param intensity The intensity of the balancing. How close to the ideal should we go from our current position (0=do not change, 1 go to ideal)
+ * @param minBudget Ensure that we use at least min_budget (in order to keep the overall rate)
+ * @returns The balanced items and the used budget.
+ */
```

Comment:
Is implementation drift going to be an issue here, i.e. is there a good reason why we don't recompute on submitting and doing that via an API call? I have a feeling this will drift apart quite quickly

Reply:
We are going to move this to FE? And just store the target sample rates?

Reply:
Ahh okay I was not aware of that.